Update BlightVeil_Kill_Tracker01.py

UntestedVibeCode opened the pull request.

BossGamer09 committed:
Update BlightVeil_Kill_Tracker01.py

UntestedVibeCode committed:
Fix rsi_handle

Remove trailing ']'

UntestedVibeCode committed:
Removed loading animation

loading animation broken

BossGamer09 committed:
Updated Code with New Base

Merged Commander Mode Code into the Base Code we are using

DefinitelySafeCode commented on BlightVeil_Kill_Tracker01.py:
```diff
@@ -1,38 +1,80 @@
+update_queue = Queue()
+
+# Ensure external sounds folder exists before doing anything else
+os.makedirs(SOUNDS_FOLDER, exist_ok=True)  # Cleaner way to create folder if missing
+
+global_ship_list = [
+    'DRAK', 'ORIG', 'AEGS', 'ANVL', 'CRUS', 'BANU', 'MISC',
+    'KRIG', 'XNAA', 'ARGO', 'VNCL', 'ESPR', 'RSI', 'CNOU',
+    'GRIN', 'TMBL', 'GAMA'
+]
 
 def resource_path(relative_path):
-    """ Get the absolute path to the resource (works for PyInstaller .exe). """
+    """ Get the absolute path to the resource (works for PyInstaller .exe and Nuitka .exe). """
     try:
+        # When running in a frozen environment (compiled executable)
         base_path = sys._MEIPASS  
     except AttributeError:
+        # When running in a normal Python environment (source code)
         base_path = os.path.abspath(".")  
     return os.path.join(base_path, relative_path)
 
+# Folder where user places their sounds (next to the .exe when running a built version)
+EXTERNAL_SOUNDS_FOLDER = os.path.join(os.getcwd(), "sounds")
+
+SOUNDS_FOLDER = resource_path("sounds")
+
+# Ensure the external sounds folder exists before doing anything else
+os.makedirs(EXTERNAL_SOUNDS_FOLDER, exist_ok=True)
+os.makedirs(SOUNDS_FOLDER, exist_ok=True)
```
Sound setup belongs in its own function that should be called at startup of the application.

DefinitelySafeCode commented on BlightVeil_Kill_Tracker01.py:
```diff
@@ -1,38 +1,80 @@
         if response.status_code == 200:
             release_data = response.json()
-            remote_version = release_data.get("tag_name", "v1.0").strip("v")
+            remote_version = release_data.get("tag_name", "v1.3").strip("v")
```
why strip "v" ?

DefinitelySafeCode commented on BlightVeil_Kill_Tracker01.py:
```diff
@@ -43,79 +85,137 @@ def check_for_updates():
+
+    # Get the list of existing files in the program's sounds folder
+    existing_files = set(os.listdir(SOUNDS_FOLDER))
+
+    try:
+        for sound_file in os.listdir(EXTERNAL_SOUNDS_FOLDER):
+            source_path = os.path.join(EXTERNAL_SOUNDS_FOLDER, sound_file)
+            target_path = os.path.join(SOUNDS_FOLDER, sound_file)
+
+            # Check if it's a file and if it's not already copied
+            if os.path.isfile(source_path) and sound_file not in existing_files:
+                shutil.copy(source_path, target_path)
+                print(f"Copied new sound: {sound_file} to {SOUNDS_FOLDER}")
+
+    except Exception as e:
+        print(f"Error copying sounds: {e}")
+
+def get_all_sounds():
+    """Fetch only .wav sounds from the sounds folder."""
+    if not os.path.exists(SOUNDS_FOLDER):
+        print("❌ Sounds folder not found!")
+        return []
+    
+    return [
+        os.path.join(SOUNDS_FOLDER, f) for f in os.listdir(SOUNDS_FOLDER) 
+        if f.endswith(".wav")
+    ]
+    
+def play_random_sound():
+    """Play a single random .wav file from the sounds folder."""
+    sounds = get_all_sounds()
+    if sounds:
+        sound_to_play = random.choice(sounds)  # Select a random sound
+        try:
+            logger.log(f"✅ Playing sound: {sound_to_play}")
+            winsound.PlaySound(sound_to_play, winsound.SND_FILENAME)  # Play the selected sound
+        except Exception as e:
+            print(f"⚠️ Error playing sound {sound_to_play}: {e}")
+    else:
+        print("❌ No .wav sound files found.")
```
Good candidate to move these to a killtracker_sounds.py file. Time to start refactoring, might as well do this one first.

DefinitelySafeCode commented on BlightVeil_Kill_Tracker01.py:
```diff
@@ -770,7 +827,214 @@ def refresh_user_list(active_users):
+            'mode': "commander"
+        }
+
+        if not api_key["value"]:
+            logger.log("Heartbeat will not be sent. Enter valid key to establish connection with Servitor...")
+            global_heartbeat_active = False
+            return
+
+        try:
+            #logger.log(f"Sending heartbeat: {json_data}")
+            response = requests.post(heartbeat_url, headers=headers, json=json_data, timeout=5)
+            response.raise_for_status()  # Raises an exception for HTTP errors
+
+            response_data = response.json()
+            #logger.log(f"Received response: {response_data}")
+
+            # Update the UI with active commanders if the response contains the key
+            if 'commanders' in response_data:
+                active_commanders = response_data['commanders']
+                # Put the updated commanders list in the queue for the GUI thread to process
+                update_queue.put(active_commanders)
+            else:
+                logger.log("No commanders found in response.")
+
+        except requests.RequestException as e:
+            logger.log(f"Error sending heartbeat: {e}")
+            global_heartbeat_active = False
+            return        
+
+def start_heartbeat_thread(logger):
+    """Start the heartbeat in a seperate thread"""
+    global global_rsi_handle
+    global global_heartbeat_active
+    if global_heartbeat_active:
+        logger.log("Already connected to commander!")
+        return
+    logger.log("Connecting to commander...")
+    thread = threading.Thread(target=post_heartbeat, args=(global_rsi_handle, logger))
+    thread.daemon = True
+    thread.start()
```
IF not a huge pain in the ass, move heartbeat code to a killtracker_commandermode.py

UntestedVibeCode committed:
Commander Mode Cleanup

Fixed colors on connect.
Prevent multiple thread runs.
Cleaned up buttons.
Made sure same user could not be added twice to the allocated forced.
Tried to clear out listboxes on disconnect.
Made disconnect join the threads.

UntestedVibeCode committed:
Fixed player updates

tkhu1 committed:
fixes to search and user allocation (#6)

* add gitignore and requirements

* fixes to search and user allocation

- remove unused libraries
- fix window size
- better search bar
- fix bottom padding of list frames
- new add all to fleet button
- remove user from allocate forces if they disconnect
- remove any dupes from heartbeat response if they exist
- sort connected users by alphabetical order
- started docs
